ft-sdk/src/auth/provider.rs

```diff
@@ -290,23 +293,57 @@ pub fn create_user(
     Ok(ft_sdk::auth::UserId(user_id))
 }
 
-/// persist the user in session and redirect to `next`
+/// Logs in a user and manages session creation or update.
+///
+/// # Arguments
+///
+/// * `conn` - Mutable reference to a `ft_sdk::Connection` to interact with the database.
+/// * `user_id` - Reference to a `ft_sdk::UserId` representing the user's ID.
+/// * `session_id` - Optional `ft_sdk::auth::SessionID` representing an existing session ID.
+///
+/// # Returns
 ///
-/// `identity`: Eg for GitHub, it could be the username. This is stored in the cookie so can be
-/// retrieved without a db call to show a user identifiable information.
+/// A `Result` containing a `ft_sdk::auth::SessionID` if the login operation is successful,
+/// or a `LoginError` if there's an issue with the login process.
 pub fn login(
     conn: &mut ft_sdk::Connection,
-    ft_sdk::UserId(user_id): &ft_sdk::UserId,
+    user_id: &ft_sdk::UserId,
+    session_id: Option<ft_sdk::auth::SessionID>,
```

Comment:
Not sure, why are we doing this? If there is a session id then user must be logged in, we do not have any other use of session right now other than to track if user is logged in or not. Why would session exist without user being logged in? If for existing session you are trying to log another user in, then that is wrong too, on logout the session cookie should be deleted and a new session should be created for next user getting logged in.

Reply:
This is added by you. I am not sure of the logic though. Please recheck and let me know what to do.

Reply:
I think we were trying to do this: “This is what django does, django creates a session as soon as you try to store anything in the session. And it re-uses the session during login (so session data set pre-login is not lost when you log-in). On logout it deletes the session.”. This will need more discussion, so let’s keep the current logic as is, I have created a discussion in the #dev channel where we can hash it out.